lib/index.ts

```diff
@@ -372,22 +374,73 @@ export default class Backburner {
+  public throttle<T>(target: T, methodName: keyof T, arg1: any, arg2: any, wait: number): Timer;
+  public throttle<T>(target: T, methodName: keyof T, arg1: any, arg2: any, arg3: any, wait: number): Timer;
+
+  // with target, with method name, with immediate
+  public throttle<T>(target: T, methodName: keyof T, wait: number, immediate: boolean): Timer;
+  public throttle<T>(target: T, methodName: keyof T, arg1: any, wait: number, immediate: boolean): Timer;
+  public throttle<T>(target: T, methodName: keyof T, arg1: any, arg2: any, wait: number, immediate: boolean): Timer;
+  public throttle<T>(target: T, methodName: keyof T, arg1: any, arg2: any, arg3: any, wait: number, immediate: boolean): Timer;
+
+  // with target, without immediate
+  public throttle(thisArg: any, method: () => void, wait: number): Timer;
+  public throttle<A>(thisArg: any, method: (arg1: A) => void, arg1: A, wait: number): Timer;
+  public throttle<A, B>(thisArg: any, method: (arg1: A, arg2: B) => void, arg1: A, arg2: B, wait: number): Timer;
+  public throttle<A, B, C>(thisArg: any, method: (arg1: A, arg2: B, arg3: C) => void, arg1: A, arg2: B, arg3: C, wait: number): Timer;
+
+  // with target, with immediate
+  public throttle(thisArg: any, method: () => void, wait: number, immediate: boolean): Timer;
+  public throttle<A>(thisArg: any, method: (arg1: A) => void, arg1: A, wait: number, immediate: boolean): Timer;
+  public throttle<A, B>(thisArg: any, method: (arg1: A, arg2: B) => void, arg1: A, arg2: B, wait: number, immediate: boolean): Timer;
+  public throttle<A, B, C>(thisArg: any, method: (arg1: A, arg2: B, arg3: C) => void, arg1: A, arg2: B, arg3: C, wait: number, immediate: boolean): Timer;
+
+  // without target, default immediate
+  public throttle(method: () => void, wait: number): Timer;
+  public throttle<A>(method: (arg1: A) => void, arg1: A, wait: number): Timer;
+  public throttle<A, B>(method: (arg1: A, arg2: B) => void, arg1: A, arg2: B, wait: number): Timer;
+  public throttle<A, B, C>(method: (arg1: A, arg2: B, arg3: C) => void, arg1: A, arg2: B, arg3: C, wait: number): Timer;
+
+  // without target, with immediate
+  public throttle(method: () => void, wait: number, immediate: boolean): Timer;
+  public throttle<A>(method: (arg1: A) => void, arg1: A, wait: number, immediate: boolean): Timer;
+  public throttle<A, B>(method: (arg1: A, arg2: B) => void, arg1: A, arg2: B, wait: number, immediate: boolean): Timer;
+  public throttle<A, B, C>(method: (arg1: A, arg2: B, arg3: C) => void, arg1: A, arg2: B, arg3: C, wait: number, immediate: boolean): Timer;
+  public throttle(targetOrThisArgOrMethod: Object | Function, ...args): Timer {
+    let target;
+    let method;
+    let immediate;
     let isImmediate;
     let wait;
 
-    if (isCoercableNumber(immediate)) {
-      wait = immediate;
+    if (args.length === 1) {
+      method = targetOrThisArgOrMethod;
+      wait = args[0];
```

Comment:
I think it should `args.pop()` here, else timer `wait` will be passed to debounced function

Reply:
Yep, you are definitely correct.  I updated to fix that and ensure it is under test...